robot/impl/local_robot.go

```diff
@@ -1194,18 +1194,18 @@ func (r *localRobot) reconfigure(ctx context.Context, newConfig *config.Config,
 	// First we mark diff.Removed resources and their children for removal.
 	processesToClose, resourcesToCloseBeforeComplete, _ := r.manager.markRemoved(ctx, diff.Removed, r.logger)
 
-	// Second we update the resource graph and stop any removed processes.
-	allErrs = multierr.Combine(allErrs, r.manager.updateResources(ctx, diff))
-	allErrs = multierr.Combine(allErrs, processesToClose.Stop())
-
-	// Third we attempt to Close resources.
+	// Second we attempt to Close resources.
 	alreadyClosed := make(map[resource.Name]struct{}, len(resourcesToCloseBeforeComplete))
 	for _, res := range resourcesToCloseBeforeComplete {
 		allErrs = multierr.Combine(allErrs, r.manager.closeResource(ctx, res))
 		// avoid a double close later
 		alreadyClosed[res.Name()] = struct{}{}
 	}
 
+	// Third we update the resource graph and stop any removed processes.
+	allErrs = multierr.Combine(allErrs, r.manager.updateResources(ctx, diff))
```

Comment:
Don't feel like closing `ProcessesToClose` in a different order is a concern, but I am not entirely sure if calling `updateResources()` before closing resources marked for removal is safe. From what I understand, it just marks new/modified resources for updates for the resource graph so should be okay but I could totally be overlooking something.